internal/util/iterator/close.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+// withClose wraps an iterator with a custom close function.
+type withClose[K, V any] struct {
+	iter  Interface[K, V]
+	close func()
+}
+
+// WithClose wraps an iterator with a custom close function.
+//
+// That function should call Close() method of the wrapped iterator.
+func WithClose[K, V any](iter Interface[K, V], close func()) Interface[K, V] {
```

Comment:
I don't understand the usage, can you provide some basic example? I've tried to came up with something locally but still doesn't understand. The close function provided by user should close the transaction, but what's the point if we do that already with InTransaction function?

Reply:
See https://github.com/FerretDB/FerretDB/pull/1769/files#r1101318420